src/main/java/cam72cam/immersiverailroading/thirdparty/CommonAPI.java

```diff
@@ -19,6 +19,33 @@
 
 public class CommonAPI {
     private final EntityRollingStock stock;
+    public static final HashMap<String, stockProperty> PropMap;
+    static  
+    {  
+        PropMap = new HashMap<String, stockProperty>();  
```

Comment:
This map is not needed.  You can interface with the enum like:
name of property:  stockProperty.toString().toLowerCase()
property from name: stockProperty.valueOf(inputName.toUpperCase())

Reply:
Could even add helper functions inside the enum definition itself

Reply:
It is used to convert String arguments from Lua script to Enum.
I think maybe the name of properties will be changed in future.
If we have this hashmap, the name and the arguments will be decoupled.
Well, if you insist...

Reply:
If that is a concern, we could change the constructor of the enum to take a string to use as an identifier and do the lookup based on that